enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/k8s/PortForwards.java

```diff
@@ -0,0 +1,235 @@
+                if (ports.isEmpty()) {
+                    NotificationUtils.showMessage("No ports found for pod: " + podItem.getName());
+                    return;
+                }
+                try {
+                    for (Integer port : ports) {
+                        if (!isPortAvailable(port)) {
+                            NotificationUtils.showErrorMessage(Bundle.PortNotFree(port));
+                            break;
+                        }
+                        LocalPortForward fwd = client.pods()
+                                .inNamespace(pod.getMetadata().getNamespace())
+                                .withName(pod.getMetadata().getName())
+                                .portForward(port, port);
+                        forwards.add(fwd);
+                        forwardItems.add(new PortForwardItem(podItem, fwd.getLocalPort(), port));
+
+                        NotificationUtils.showMessage(Bundle.ForwardingPorts(port.toString(), podItem.getName()));
+                    }
+                    latches.put(podItem.getName(), latch);
+                    List<PortForwardItem> oldValue = activePortForwards.put(podItem.getName(), forwardItems);
+                    firePropertyChange(podItem, oldValue, forwardItems);
+                    latch.await();
+                    for (PortForward forward : forwards) {
+                        forward.close();
+                    }
+                    activePortForwards.remove(podItem.getName());
+                    firePropertyChange(podItem, forwardItems, oldValue);
+                } catch (InterruptedException | IOException ex) {
+                    Exceptions.printStackTrace(ex);
+                } catch (IllegalStateException e) {
+                    NotificationUtils.showErrorMessage(e.getMessage());
+                } finally {
+                    handle.finish();
+                }
+
+            });
+        });
+    }
+
+    /**
+     * Retrieves the list of active port forwards for the specified Pod.
+     *
+     * @param podName The name of the Pod.
+     * @return A list of active {@code PortForward} instances for the Pod, or an
+     * empty list if none are found.
+     */
+    public List<PortForwardItem> getActivePortForwards(String podName) {
+        return activePortForwards.getOrDefault(podName, Collections.emptyList());
+    }
+
+    /**
+     * Stops and closes all active port forwards for the specified Pod.
+     *
+     * <p>
+     * Resources associated with the port forwarding are released, and the port
+     * forward is removed from the active list.</p>
+     *
+     * @param podName The name of the Pod for which port forwarding should be
+     * stopped.
+     */
+    public void closePortForward(String podName) {
+        CountDownLatch latch = latches.get(podName);
+        if (latch != null) {
+            latch.countDown();
+        }
+        activePortForwards.remove(podName);
```

Comment:
I suggest not to remove activePortForwards/latches here, but just request the shutdown using countDown, and let the closing code that call portForward.close() to remove the forward after it is closed (or fails to close).